packages/amplify-auth-sdk/src/stores/secure-store.js

```diff
@@ -19,19 +25,64 @@ export default class SecureStore extends FileStore {
 	/**
 	 * Loads the `keytar` library and initializes the token file.
 	 *
-	 * @param {Object} [opts] - Various options.
+	 * @param {Object} opts - Various options.
+	 * @param {String} opts.homeDir - The path to the home directory containing the `lib`
+	 * directory where `keytar` is located.
 	 * @param {String} [opts.secureServiceName="Axway AMPLIFY Auth"] - The name of the consumer
      * using this library.
 	 * @access public
 	 */
 	constructor(opts = {}) {
+		if (!opts || typeof opts !== 'object') {
+			throw E.INVALID_ARGUMENT('Expected opts to be an object');
+		}
+
+		const { homeDir } = opts;
+		if (!homeDir || typeof homeDir !== 'string') {
+			throw E.INVALID_PARAMETER('Secure store requires the home directory to be specified');
+		}
+
+		const prefix = path.join(homeDir, 'lib', 'keytar', `${process.platform}_${process.arch}_${process.versions.modules}`);
+		const keytarPath = path.join(prefix, 'node_modules', 'keytar');
+		log(`Attempting to load keytar: ${highlight(keytarPath)}`);
+
 		let keytar;
 
 		try {
-			keytar = require('keytar');
+			keytar = require(keytarPath);
 		} catch (e) {
-			/* istanbul ignore next */
-			throw E.KEYTAR_NOT_FOUND('"keytar" package not found, is it installed?');
+			if (e.code !== 'MODULE_NOT_FOUND') {
+				log(e);
+				throw e;
+			}
+
+			const args = [ 'install', 'keytar@4.4.1', '--no-save', '--production', '--prefix', prefix ];
```

Comment:
Do we want the version in source? Maybe read the package.json and pull that version?

Reply:
I did spend a few minutes thinking about versioning keytar and ultimately I decided to lock down the version of `keytar`. Otherwise I'd have to build a mechanism that checks the version of the installed `keytar`, then do a `semver` on it to see if there's a new version.

I mentioned this as a caveat in the ticket, so unless someone has a strong opinion fixing it now, I'd rather defer a better upgrade solution and get these changes out sooner than later.

Reply:
Fixed! I added a `"keytar"` property to the `package.json`.